language-support/ts/codegen/src/Main.hs

```diff
@@ -168,48 +168,51 @@ genDefDataType curModName tpls def = case unTypeConName (dataTypeCon def) of
             map ("  " <>) (onHead ("decoder: " <>) serDesc) ++
             ["});"]
 
-genType :: ModuleName -> Type -> (T.Text, T.Text)
+genType :: ModuleName -> Type -> ((T.Text, T.Text), Set.Set ModuleRef)
 genType curModName = go
   where
     go = \case
-        TVar v -> dup (unTypeVarName v)
-        TUnit -> ("{}", "daml.Unit")
-        TBool -> ("boolean", "daml.Bool")
-        TInt64 -> dup "daml.Int"
-        TDecimal -> dup "daml.Decimal"
-        TNumeric _ -> dup "daml.Numeric"  -- TODO(MH): Figure out what to do with the scale.
-        TText -> ("string", "daml.Text")
-        TTimestamp -> dup "daml.Time"
-        TParty -> dup "daml.Party"
-        TDate -> dup "daml.Date"
+        TVar v -> (dup (unTypeVarName v), Set.empty)
```

Comment:
Manually writing the code for detecting the references seems a bit repetitive and brittle since it’s easy to forget to recurse somewhere. Is there a reason why we cannot throw some traversal at this that takes care of giving us all module references?